projects/bms_carrier/test/test_killswitch.c

```diff
@@ -1,6 +1,70 @@
-#include "gpio_it.h"
 #include "killswitch.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "bms.h"
+#include "exported_enums.h"
 #include "log.h"
+#include "test_helpers.h"
+
+static uint8_t s_bps_fault_mask = 0;
+static bool s_bps_fault_clear = false;
+static uint8_t s_fault_bps_calls = 0;
+static GpioState s_gpio_state_get = NUM_GPIO_STATES;
+static GpioItCallback s_registered_callback = NULL;
+static void *s_registered_context = NULL;
+
+static DebouncerStorage s_killswitch_storage = { 0 };
+
+StatusCode TEST_MOCK(fault_bps)(uint8_t fault_bitmask, bool clear) {
+  s_bps_fault_mask = fault_bitmask;
+  s_bps_fault_clear = clear;
+  s_fault_bps_calls++;
+  return STATUS_CODE_OK;
+}
+
+StatusCode TEST_MOCK(gpio_get_state)(const GpioAddress *address, GpioState *state) {
+  *state = s_gpio_state_get;
+  return STATUS_CODE_OK;
+}
+
+StatusCode TEST_MOCK(debouncer_init_pin)(DebouncerStorage *debouncer, const GpioAddress *address,
+                                         GpioItCallback callback, void *context) {
+  s_registered_callback = callback;
+  s_registered_context = context;
+  return STATUS_CODE_OK;
+}
```

Comment:
why do you have to mock this? If it's because an x86 is missing, we should probably add that (or add a ticket and TODO works too)

Reply:
I have no idea why I did this, just removed this mock. Debouncer is platform agnostic.